internal/controllers/image/upload.go

```diff
@@ -43,76 +45,78 @@ func (r *orcImageReconciler) hashVerifier(ctx context.Context, orcImage *orcv1al
 		} else {
 			log.V(logging.Info).Info("download hash verification failed", "expected", expectedValue, "got", hash)
 			msg := "download hash verification failed. got: " + hash
-			r.recorder.Eventf(orcImage, corev1.EventTypeWarning, "HashVerificationFailed", msg)
+			// actuator.recorder.Eventf(orcImage, corev1.EventTypeWarning, "HashVerificationFailed", msg)
 			return errors.New(msg)
 		}
 		return nil
 	}
 }
 
-func (r *orcImageReconciler) downloadProgressReporter(ctx context.Context, orcImage *orcv1alpha1.Image, glanceImage *images.Image, contentLength int64) progressReporter {
-	log := ctrl.LoggerFrom(ctx)
+func (actuator imageActuator) downloadProgressReporter(ctx context.Context, orcImage *orcv1alpha1.Image, glanceImage *images.Image, contentLength int64) progressReporter {
+	// log := ctrl.LoggerFrom(ctx)
 
-	var ofTotal string
-	if contentLength > 0 {
-		ofTotal = fmt.Sprintf("/%dMB", int(contentLength/1024/1024))
-	}
+	// var ofTotal string
+	// if contentLength > 0 {
+	// 	ofTotal = fmt.Sprintf("/%dMB", int(contentLength/1024/1024))
+	// }
 
 	interval := 10 * time.Second
 	nextUpdate := time.Now().Add(interval)
 	return func(progress int64) {
 		if time.Now().After(nextUpdate) {
-			msg := fmt.Sprintf("Downloaded %dMB"+ofTotal, int(progress/1024/1024))
-			err := r.updateStatus(ctx, orcImage, withResource(glanceImage),
-				withProgressMessage(downloadingMessage(msg, orcImage)))
-			if err != nil {
-				// Failure to update status here is not fatal
-				log.Error(err, "Error writing status during image upload")
-			}
+			// msg := fmt.Sprintf("Downloaded %dMB"+ofTotal, int(progress/1024/1024))
+			// err := r.updateStatus(ctx, orcImage, withResource(glanceImage),
+			// 	withProgressMessage(downloadingMessage(msg, orcImage)))
+			// if err != nil {
+			// 	// Failure to update status here is not fatal
+			// 	log.Error(err, "Error writing status during image upload")
+			// }
 			nextUpdate = time.Now().Add(interval)
 		}
 	}
 }
 
-func (r *orcImageReconciler) uploadImageContent(ctx context.Context, orcImage *orcv1alpha1.Image, imageClient osclients.ImageClient, glanceImage *images.Image) (err error) {
+func (actuator imageActuator) uploadImageContent(ctx context.Context, orcImage *orcv1alpha1.Image, glanceImage *images.Image) ([]progress.ProgressStatus, error) {
 	log := ctrl.LoggerFrom(ctx)
 	log.V(logging.Info).Info("Uploading image content")
 
+	var waitEvents []progress.ProgressStatus
```

Comment:
I don't think this is relevant here because we don't plan to update the Progressing condition based on the outcome of this function. At least we aren't currently doing that.

We should have set Progressing to something like 'Waiting for image data' in the previous reconcile. In this reconcile we're going to write to a different condition, maybe `ImageDataReady`? This other condition is going to contain download progress when it is False.